contrib/babelfishpg_tsql/sql/sys_views.sql

```diff
@@ -2102,22 +2098,41 @@ SELECT CAST('TSQL Default TCP' AS sys.sysname) AS name
   , CAST(0 AS sys.bit) AS is_admin_endpoint;
 GRANT SELECT ON sys.endpoints TO PUBLIC;
 
-create or replace view sys.index_columns
-as
-select i.indrelid::integer as object_id
-  , CAST(CASE WHEN i.indisclustered THEN 1 ELSE 1+row_number() OVER(PARTITION BY c.oid) END AS INTEGER) AS index_id
-  , a.attrelid::integer as index_column_id
-  , a.attnum::integer as column_id
-  , a.attnum::sys.tinyint as key_ordinal
-  , 0::sys.tinyint as partition_ordinal
-  , 0::sys.bit as is_descending_key
-  , 1::sys.bit as is_included_column
-from pg_index as i
-inner join pg_catalog.pg_attribute a on i.indexrelid = a.attrelid
-inner join pg_class c on i.indrelid = c.oid
-inner join sys.schemas sch on sch.schema_id = c.relnamespace
-where has_schema_privilege(sch.schema_id, 'USAGE')
-and has_table_privilege(c.oid, 'SELECT,INSERT,UPDATE,DELETE,TRUNCATE,TRIGGER');
+CREATE OR REPLACE VIEW sys.index_columns
+AS
+SELECT
+    CAST(i.indrelid AS INT) AS object_id,
+    -- should match index_id of sys.indexes 
+    CAST(CASE
+            WHEN i.indisclustered THEN 1
+            ELSE (SELECT COUNT(*)+2 FROM pg_index ind WHERE ind.indrelid = i.indrelid AND ind.indexrelid < i.indexrelid AND ind.indisclustered = false)
```

Comment:
Use Materialize CTE if worst case we want to go with a subquery approach
Materialize CTE will not be folded multiple times in a complex query-plan which joins index_columns with other TSQL views